app/routers/quizzes.py

```diff
@@ -92,6 +93,75 @@ async def get_quiz(quiz_id: str):
+
+        if "metadata" in quiz and quiz["metadata"] is not None:
+            if "quiz_type" in quiz["metadata"]:
+                if quiz["metadata"]["quiz_type"] == QuizType.omr.value:
+                    question_set_ids = [
+                        question_set["_id"] for question_set in quiz["question_sets"]
+                    ]
+
+                    # find questions with given question set ids
+                    # count number of options for each question in a qset id
+                    # group them together into an optionsArray
+                    options_count_across_sets = list(
+                        client.quiz.questions.aggregate(
+                            [
+                                {
+                                    "$match": {
+                                        "question_set_id": {"$in": question_set_ids}
+                                    }
+                                },
+                                {
+                                    "$sort": {"_id": 1}
+                                },  # sort questions based on question_id
+                                {
+                                    "$project": {
+                                        "_id": 0,
+                                        "question_set_id": "$question_set_id",
+                                        "number_of_options": {"$size": "$options"},
+                                    }
+                                },
+                                {
+                                    "$group": {
+                                        "_id": "$question_set_id",
+                                        "options_count_per_set": {
+                                            "$push": "$number_of_options"
+                                        },
+                                    }
+                                },
+                                {
+                                    "$sort": {"_id": 1}
+                                },  # sort sets based on question_set_id
+                                {"$project": {"_id": 0, "options_count_per_set": 1}},
+                            ]
+                        )
+                    )
+
+                    for question_set_index, question_set in enumerate(
+                        quiz["question_sets"]
+                    ):
+
+                        updated_subset_without_details = []
+                        options_count_per_set = options_count_across_sets[
+                            question_set_index
+                        ]["options_count_per_set"]
+                        for question_index, question in enumerate(
+                            question_set["questions"]
+                        ):
+                            if question_index < settings.subset_size:
+                                continue
+
+                            # options_count will be zero for subbjective/numerical questions
```

Comment:
Why? This is a behaviour of the current test design? Maybe the quiz engine shouldn't depend on it, no?

Reply:
Do you mean `options_count` being zero for subjective/numerical? Quiz Engine doesn't depend on it as such. The above mongo aggregate will set it to zero anyway, as in, Line158 `options_count_per_set[question_index]` would be zero for a subjective/numerical question.

The reason we are doing this aggregate: our subset pattern logic doesn't send number-of-options for questions beyond subset size. Omr view requires info of number-of-options for all questions to render its components. 

Had we not done this aggregate, questions beyond subset size would look blank in `OmrModal`, not displaying any options (bubbles).